lib/tests/test_merged_tree.rs

```diff
@@ -50,7 +50,12 @@ fn diff_stream_equals_iter(tree1: &MergedTree, tree2: &MergedTree, matcher: &dyn
 
 #[test]
 fn test_from_legacy_tree() {
-    let test_repo = TestRepo::init();
+    // TODO: Find out why this test does not works with tree-level conflicts
```

Comment:
I think this test is explicitly about checking that an existing legacy tree can be read, but if we enable tree-level conflicts by defaults then the will be no legacy tree, right?